### DISCUSSION TITLE
Optimal Settings for Voice Assistant Operation (Guidance Needed)

### DISCUSSION THREAD

**Discussion by mllockwood** [dinki/View-Assist#11] [Q&A]
I have loaded RTPMic on my ThinkSmart devices running Android 11.  The app is setup and configured per the setup instructions/guide.  I have configured the Stream Assist integration in Home Assistant, and added each of my ThinkSmart devices.  I have the Voice Assistant configuration in Home Assistant configured as well.  However, I am having very poor results using the ThinkSmart devices as voice satellites.  Previously, I was running RPi Zero devices, configured with local wake word detection, and those performed fantastic (aside from constant false wake word triggers).  The issue I'm having is that I say the wake word "Hey Jarvis", and shortly thereafter I hear the STT start media MP3 file played through the ThinkSmart's speaker.  I then say the phrase or command.  Then, nothing happens.  Sometimes, after about 30 seconds or so, I will get a response like "I"m sorry, I was unable to process your request" or something along the lines of that.

Would some of you mind sharing your settings with me, that you have working?  

- Gain settings in RTPMic (I have mine set to +9dB)
- Which Conversation Agent you're using (I have tried ChatGPT and Home Assistant)
- Which Speech to Text engine you're using (I have tried Faster Whisper and Home Assistant Cloud)
- Which Text to Speech engine you're using (I have tried Piper and Home Assistant Cloud)
- What wake word are you using?
- If you are using ChatGPT, what LLM and settings are you using?

**Comment by mngarchow** [comment]
Here are my successful settings. 

RTPmic is set to +9dB
Home Assistant (English)
faster-whisper (English)
piper (en_US) with danny(low) voice
openwakeword with hey jarvis

I also have the STT start mp3. If you are getting a long delay and then the "I'm sorry..." message, that seems to me like it doesn't hear you.

**Comment by dinki** [comment]
> I also have the STT start mp3. If you are getting a long delay and then the "I'm sorry..." message, that seems to me like it doesn't hear you.

Totally agree with @mngarchow  on this.  I had a similar issue while setting up an ESP32 satellite.  Wakeword detect worked fine but it would struggle to understand me.  After a whole lot of chasing problems that didn't exist, I came across someone else that had the same problem and found the solution was to increase the gain on the mic.

This is anecdotal but I think it applies.  In my quest to solve, I did also read someone saying that they were somehow able to save raw audio and when they played it back they could tell right away what the problem was.  I don't know how to do that or if it is even possible with Stream Assist.

**Reply by mllockwood** [reply]
Following the RTPMic installation and configuration instructions, it mentions configuring the app to use port 5555, with the broadcast address of 255.255.255.255.  Should all of the ThinkSmart devices use port 5555, or should each have it's own unique port number?  Pros/cons of running them all the same versus unique?

**Reply by dinki** [reply]
All RTPMic installs should be configured the same way.  The software is running on port 55555 (five fives) on each device so there is no reason to change them.  Nothing else is running on those ports on the tablets so no worries.  If you want, I'm sure you could change them and just ensure they match on the Stream Assist configuration but why bother?
